sys/net/application_layer/dhcpv6/client.c

```diff
@@ -1042,10 +1042,14 @@ static bool _parse_reply(uint8_t *rep, size_t len, uint8_t request_type)
             default:
                 break;
         }
+        /* TP-Link DHCP server uses this as an end marker? */
+        if (!byteorder_ntohs(opt->type)) {
+            break;
+        }
```

Comment:
The question is whether this might in fact be standard compliant (as both Windows and Linux have no issue with getting an address from that router) but I did not read up on the RFC yet.

I think the likelihood of us doing something wrong here is greater than a vendor shipping devices that are *obviously working* with non-RIOT clients in the wild.

Reply:
> The question is whether this might in fact be standard compliant (as both Windows and Linux have no issue with getting an address from that router)

Oh, okay, that is interesting! I searched through the RFC yesterday a bit but couldn't find anything in this regard yet.

> I think the likelihood of us doing something wrong here is greater than a vendor who is shipping devices that are obviously working with non-RIOT clients in the wild.

That's a very good point ;) Then I guess the workaround makes sense for now. Should we open an issue after merging to remind us to investigate this further if we can't find anything in the RFC?

Reply:
I just tried some other serves, with KEA I get

```
DHCPv6 client: received REPLY
type: 1
type: 2
type: 25
type: 25
type: 0
```

and with my Fritz!Box

```
2022-03-22 21:11:29,940 # DHCPv6 client: received REPLY
2022-03-22 21:11:29,941 # type: 82
2022-03-22 21:11:29,942 # type: 1
2022-03-22 21:11:29,942 # type: 2
2022-03-22 21:11:29,943 # type: 7
2022-03-22 21:11:29,944 # type: 23
2022-03-22 21:11:29,945 # type: 86
2022-03-22 21:11:29,946 # type: 25
2022-03-22 21:11:29,946 # type: 0
```

So there is indeed always a 0 option at the end.

Reply:
Thank you for investigating! It is a bit odd, though, that the use of option 0 is not specified anywhere.